node/src/components/storage/tests.rs

```diff
@@ -929,30 +982,90 @@ fn should_hard_reset() {
     );
 
-    // Initialize a new storage with a hard reset to era 2, effectively hiding blocks 6 and 7.
-    let mut storage = storage_fixture_with_hard_reset(&harness, EraId::from(2));
-    // Check the highest block is #5.
-    assert_eq!(
-        Some(blocks[blocks_count - blocks_per_era].clone()),
-        get_highest_block(&mut harness, &mut storage)
-    );
+    // The closure doing the actual checks.
+    let mut check = |reset_era: usize| {
+        // Initialize a new storage with a hard reset to the given era, deleting blocks from that
+        // era onwards.
+        let mut storage = storage_fixture_with_hard_reset(&harness, EraId::from(reset_era as u64));
+
+        // Check highest block is the last from the previous era, or `None` if resetting to era 0.
+        let highest_block = get_highest_block(&mut harness, &mut storage);
+        if reset_era > 0 {
+            assert_eq!(
+                blocks[blocks_per_era * reset_era - 1],
+                highest_block.unwrap()
+            );
+        } else {
+            assert!(highest_block.is_none());
+        }
 
-    // Initialize a new storage with a hard reset to era 1, effectively hiding blocks 3-7.
-    let mut storage = storage_fixture_with_hard_reset(&harness, EraId::from(1));
-    // Check the highest block is #2.
-    assert_eq!(
-        Some(blocks[blocks_count - (2 * blocks_per_era)].clone()),
-        get_highest_block(&mut harness, &mut storage)
-    );
+        // Check deleted blocks can't be retrieved.
+        for (index, block) in blocks.iter().enumerate() {
+            let result = get_block(&mut harness, &mut storage, *block.hash());
+            let should_get_block = index < blocks_per_era * reset_era;
+            assert_eq!(should_get_block, result.is_some());
+        }
 
-    // Initialize a new storage with a hard reset to era 0, effectively hiding all blocks.
-    let mut storage = storage_fixture_with_hard_reset(&harness, EraId::from(0));
-    // Check the highest block is `None`.
-    assert!(get_highest_block(&mut harness, &mut storage).is_none());
+        // Check signatures of deleted blocks can't be retrieved.
+        for (index, block) in blocks.iter().enumerate() {
+            let result = get_block_signatures(&mut harness, &mut storage, *block.hash());
+            let should_get_sigs = index < blocks_per_era * reset_era;
+            assert_eq!(should_get_sigs, result.is_some());
+        }
+
+        // Check execution results in deleted blocks have been removed.
+        for (index, deploy) in deploys.iter().enumerate() {
+            let (_deploy, metadata) =
+                get_deploy_and_metadata(&mut harness, &mut storage, *deploy.id()).unwrap();
+            let should_have_exec_results = index < blocks_per_era * reset_era;
+            assert_eq!(
+                should_have_exec_results,
+                !metadata.execution_results.is_empty()
+            );
+        }
+    };
```

Comment:
All of the 3 checks above follow the same pattern, they could be extracted to a method.

Reply:
I don't see a way to make that more succinct than what's here already.  Each check is just 3 lines, and the only thing common to all 3 is the variable set to `index < blocks_per_era * reset_era`.

Reply:
They are all basically the same code with the exception for the `get_*` method the second element in `assert_eq!` call.

Reply:
If you don't think this is worth changing then I'm fine with that. It's not a blocker.